store/src/java/com/zimbra/cs/service/mail/SendMsg.java

```diff
@@ -133,9 +137,27 @@ public Element handle(Element request, Map<String, Object> context, QName respQn
                String identityId = msgElem.getAttribute(MailConstants.A_IDENTITY_ID, null);
                String dataSourceId = msgElem.getAttribute(MailConstants.A_DATASOURCE_ID, null);
                String draftId = msgElem.getAttribute(MailConstants.A_DRAFT_ID, null);
-               ItemId iidDraft = draftId == null ? null : new ItemId(draftId, zsc);
                boolean sendFromDraft = msgElem.getAttributeBool(MailConstants.A_SEND_FROM_DRAFT, false);
 
+               // get the identity by id and check if the message is send by using persona for other account
+               // As the draft is always part of  auth user's context, append the authuser accountId with draftId
+               // so that the draft will be deleted when message sent successfully in case of persona.
+               // because in case of sendAs and sendOnBehalfOf the draftId pattern is accountId:itemId.
+               if (identityId != null) {
+                   Identity identity = Provisioning.getInstance().get(zsc.getAuthToken().getAccount(), Key.IdentityBy.id, identityId);
+                   if (identity != null && zsc.getAuthToken().getAccount() != null && !zsc.getAuthToken().isAdmin()
```

Comment:
can separate zsc.getAuthToken().getAccount() into a variable, its being used multiple times within the condition

Reply:
Better would be a utility method.